### PR TITLE
Core/Misc: Update locales_gossip_menu_option to simple system

### PR DESCRIPTION
**Pull request by ChazyTheBest**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

- Update locales_gossip_menu_option to simple system
- Update gossip_menu to keep consistency
- Update gossip_menu_option to keep consistency

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** None

**Tests performed:** (Does it build, tested in-game, etc.)

- [x] Builds

### CODE REVIEW

**Comment by DDuarte** [comment]
Gossips will change to this soon: https://pastebin.com/sCRCVHBS
Should we wait before merging this PR?

**Comment by ChazyTheBest** [comment]
No problem, if you all prefer `TextID` over `OptionIndex` I can change it in a moment. And the other change I missed `OptionBroadcastTextID` to `OptionBroadCastTextID`.

File: src/server/game/Entities/Creature/GossipDef.cpp [hunk 1]
```diff
@@ -85,37 +85,37 @@ void GossipMenu::AddMenuItem(uint32 menuId, uint32 menuItemId, uint32 sender, ui
...
         /// Find the one with the given menu item id.
-        if (itr->second.OptionIndex != menuItemId)
+        if (itr->second.TextID != menuItemId)
             continue;
 
         /// Store texts for localization.
         std::string strOptionText, strBoxText;
-        BroadcastText const* optionBroadcastText = sObjectMgr->GetBroadcastText(itr->second.OptionBroadcastTextId);
-        BroadcastText const* boxBroadcastText = sObjectMgr->GetBroadcastText(itr->second.BoxBroadcastTextId);
+        BroadCastText const* optionBroadCastText = sObjectMgr->GetBroadCastText(itr->second.OptionBroadCastTextID);
```
Comment (Shauren) [inline src/server/game/Entities/Creature/GossipDef.cpp]: whats up with the random capitalizing in middle of the word "broadcast"?

Comment (malcrom) [inline src/server/game/Entities/Creature/GossipDef.cpp]: He's Spanish. Must have thought it was two words.

Comment (ChazyTheBest) [inline src/server/game/Entities/Creature/GossipDef.cpp]: it's what the changes proposed by malcrom looks like https://pastebin.com/L42KCeB6

**Pull request merged**